aiounifi/models/traffic_rule.py

```diff
@@ -0,0 +1,187 @@
+"""Traffic rules as part of a UniFi network."""
+
+from dataclasses import dataclass
+from typing import Any, NotRequired, Self, TypedDict
+
+import orjson
+
+from .api import ApiItem, ApiRequest, TypedApiResponse
+
+
+class BandwidthLimit(TypedDict):
+    """Bandwidth limit type definition."""
+
+    download_limit_kbps: int
+    enabled: bool
+    upload_limit_kbps: int
+
+
+class PortRange(TypedDict):
+    """Port range type definition."""
+
+    port_start: int
+    port_stop: int
+
+
+class IPAddress(TypedDict):
+    """IP Address for which traffic rule is applicable type definition."""
+
+    ip_or_subnet: str
+    ip_version: str
+    port_ranges: list[PortRange]
+    ports: list[int]
+
+
+class IPRange(TypedDict):
+    """IP Range type definition."""
+
+    ip_start: str
+    ip_stop: str
+    ip_version: str
+
+
+class Schedule(TypedDict):
+    """Schedule to enable/disable traffic rule type definition."""
+
+    date_end: str
+    date_start: str
+    mode: str
+    repeat_on_days: list[str]
+    time_all_day: bool
+    time_range_end: str
+    time_range_start: str
+
+
+class TargetDevice(TypedDict):
+    """Target device to which the traffic rule applies."""
+
+    client_mac: NotRequired[str]
+    network_id: NotRequired[str]
+    type: str
+
+
+class TypedTrafficRule(TypedDict):
+    """Traffic rule type definition."""
+
+    _id: str
+    action: str
+    app_category_ids: list[str]
+    app_ids: list[str]
+    bandwidth_limit: BandwidthLimit
+    description: str
+    domains: list[str]
+    enabled: bool
+    ip_addresses: list[IPAddress]
+    ip_ranges: list[IPRange]
+    matching_target: str
+    network_ids: list[str]
+    regions: list[str]
+    schedule: Schedule
+    target_devices: list[TargetDevice]
+
+
+@dataclass
+class TrafficRuleRequest(ApiRequest):
```

Comment:
```suggestion
class TrafficRuleRequest(ApiRequestV2):
```

How about we do an ApiRequestV2 in models.api that inherits from ApiRequest and add the general changes in this class to that one? Would make the class look just like the other ones

Reply:
Good suggestion, much cleaner and nicer. I added it in.